tests/lru_cache_test.py

```diff
@@ -0,0 +1,138 @@
+      self.assertEqual(cache.get("cache-a"), b"a")
+      self.assertEqual(set(path.glob("cache-*")), {path / "cache-a"})
+
+      cache.put("cache-b", b"b")
+      self.assertEqual(cache.get("cache-a"), b"a")
+      self.assertEqual(cache.get("cache-b"), b"b")
+      self.assertEqual(set(path.glob("cache-*")), {path / "cache-a", path / "cache-b"})
+
+  def test_put_empty_value(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = pathlib.Path(tmpdir)
+      cache = Impl(path, max_size=-1)
+
+      cache.put("cache-a", b"")
+      self.assertEqual(cache.get("cache-a"), b"")
+
+  def test_put_empty_key(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = pathlib.Path(tmpdir)
+      cache = Impl(path, max_size=-1)
+
+      with self.assertRaisesRegex(ValueError, r"key cannot be empty"):
+        cache.put("", b"a")
+
+  def test_eviction(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = pathlib.Path(tmpdir)
+      cache = Impl(path, max_size=2)
+
+      cache.put("cache-a", b"a")
+      cache.put("cache-b", b"b")
+
+      # `sleep()` is necessary to guarantee that `cache-b`"s timestamp is strictly greater than `cache-a`"s
+      time.sleep(1)
+      cache.get("cache-b")
+
+      # write `cache-c`, evict `cache-a`
+      cache.put("cache-c", b"c")
+      self.assertEqual(set(path.glob("cache-*")), {path / "cache-b", path / "cache-c"})
+
+      # calling `get()` on `cache-b` makes `cache-c` least recently used
+      time.sleep(1)
+      cache.get("cache-b")
+
+      # write `cache-d`, evict `cache-c`
+      cache.put("cache-d", b"d")
+      self.assertEqual(set(path.glob("cache-*")), {path / "cache-b", path / "cache-d"})
+
+  def test_eviction_with_empty_value(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = pathlib.Path(tmpdir)
+      cache = Impl(path, max_size=1)
+
+      cache.put("cache-a", b"a")
+
+      # write `cache-b` with length 0
+      # eviction should not happen even though the cache is full
+      cache.put("cache-b", b"")
+      self.assertEqual(set(path.glob("cache-*")), {path / "cache-a", path / "cache-b"})
+
+      # calling `get()` on `cache-a` makes `cache-b` least recently used
+      time.sleep(1)
+      cache.get("cache-a")
+
+      # writing `cache-c` should result in evicting the
+      # least recent used file (`cache-b`) first,
+      # but this is not sufficient to make room for `cache-c`,
+      # so `cache-a` should be evicted as well
+      cache.put("cache-c", b"c")
+      self.assertEqual(set(path.glob("cache-*")), {path / "cache-c"})
+
+  def test_existing_cache_dir(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      path = pathlib.Path(tmpdir)
+      cache = Impl(path, max_size=2)
+
+      cache.put("cache-a", b"a")
+
+      # simulates reinitializing the cache in another process
+      del cache
+      cache = Impl(path, max_size=2)
+
+      self.assertEqual(cache.get("cache-a"), b"a")
```

Comment:
Can you also add eviction here, to make sure the LRU policy survives process restarts? Like insert b, touch a, and then insert c and make sure b was deleted

Reply:
Done